configure

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+#
+# Copyright(c) 2012-2019 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+MODULE_FILE=test_mod.c
+OBJ_MOD=test_mod.o
+KERN_VER=`uname -r`
+PWD=`pwd`
+NPROC=`nproc`
+DEFINE_FILE=modules/generated_defines.h
+
+add_licence(){
+cat > $DEFINE_FILE << EOF
+/*
+* Copyright(c) 2012-2019 Intel Corporation
+* SPDX-License-Identifier: BSD-3-Clause-Clear
+*/
+
+EOF
+}
+
+add_define() {
+	echo "#define "$1 $2 >> $DEFINE_FILE
+}
+
+add_function() {
+	for arg in "$@"; do
+		echo -e "$arg" >> $DEFINE_FILE
+	done
+}
+
+compile_module(){
+	if [ $# -gt 1 ]
+	then
+		INCLUDE="#include <$2>"
+	else
+		INCLUDE=""
+	fi
+
+############# TEST MODULE #############
+cat > $MODULE_FILE << EOF
+#include <linux/module.h>
+#include <linux/kernel.h>
+$INCLUDE
+
+int init_module(void) {
+	$1;
+	return 0;
+}
+void cleanup_module(void) {};
+EOF
+#######################################
+
+	make -C /lib/modules/$KERN_VER/build M=$PWD modules\
+		obj-m=$OBJ_MOD -j$NPROC &> /dev/null
+
+	local ret=$?
+	if [ $ret -eq 0 ]; then
+		make -j$NPROC -C /lib/modules/$KERN_VER/build M=$PWD clean\
+			obj-m=test.o &> /dev/null
+	fi
+
+	return $ret
+}
+
+kernel_not_supp_fail() {
+	echo "Current kernel not supported - test module build failed!"
+	rm $DEFINE_FILE
+	exit 1
+}
+
+rm -f $DEFINE_FILE
+
+add_licence
+
+compile_module "part_round_stats(1, 1)" "linux/genhd.h"
+if [ $? -eq 0 ]; then
+	add_define "CAS_PART_ROUND_STATS(q, cpu, part)"	"part_round_stats(cpu, part)"
+else
+	compile_module "part_round_stats(NULL, 1, 1)" "linux/genhd.h"
+	if [ $? -eq 0 ]; then
+		add_define "CAS_PART_ROUND_STATS(q, cpu, part)"\
```

Comment:
I suggest moving actual macro definitions to source code. For example instead of 
     add_define "CAS_PART_ROUND_STATS(q, cpu, part) part_round_stats(q, cpu, part)" 
define just
      add_define "CAS_PART_ROUND_STATS_3_ARG"
.. and then in CAS source code (linux_kernel_version.h ?):
      #include "generated_defines.h" 
      .....
      #if defined CAS_PART_ROUND_STATS_3_ARG
           #define CAS_PART_ROUND_STATS(q, cpu, part) part_round_stats(q, cpu, part)
      #endif